core/src/main/java/org/ldaptive/pool/PruneStrategy.java

```diff
@@ -2,14 +2,19 @@
 package org.ldaptive.pool;
 
 import java.time.Duration;
-import java.util.function.Function;
+import java.util.Iterator;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
 
 /**
- * Provides an interface for pruning connections from the pool.
+ * Provides an interface for pruning connections from the pool. The list of predicates provided by the strategy are used
+ * against the entire pool in order. So that connections pruned by the first predicate are no longer available when the
+ * second predicate is used. In this fashion, a prune strategy can order any number of predicates, each of which can
+ * inspect the pool and make a determination whether to prune a connection.
  *
  * @author  Middleware Services
  */
-public interface PruneStrategy extends Function<PooledConnectionProxy, Boolean>
+public interface PruneStrategy extends Consumer<Supplier<Iterator<PooledConnectionProxy>>>
```

Comment:
At first glance this looks complex, but upon review of the abstract class it seems pretty natural: give me some connections to iterate over and remove the ones I don't like via `iterator#remove`. I think hiding the predicates from the public API is a notable improvement in clarity.